src/interface/src/app/map/map.component.ts

```diff
@@ -646,6 +667,14 @@ export class MapComponent implements AfterViewInit, OnDestroy, OnInit, DoCheck {
     );
   }
 
+  private showAreaTooComplexError() {
+    this.matSnackBar.open(
+      '[Error] your shapefile is too complex!',
```

Comment:
Everything looks straightforward, so I'm only concerned that the tone of this message feels a little abrupt and maybe mildly impolite. I know it's a snackbar so it should be short, so perhaps UX/Product have suggestions?

It might also be better UX to show a hint of what to fix --e.g., "Unable to process this shapefile. Your shapefile may be too large or have too many features".

Reply:
I like that! thanks!